File: Proyecto1/containers/low_py/low.py
from flask import Flask
import time
import threading
import random
import math

app = Flask(__name__)

# Función para consumir CPU de manera menos intensiva
def consume_cpu():
    while True:
        result = 0
        for i in range(1000):  # Reducción en la cantidad de cálculos
            result += math.sqrt(i) * math.sin(i)
        time.sleep(0.1)  # Pausa para reducir el uso continuo de CPU

# Función para consumir RAM de manera menos intensiva
def consume_ram():
    global ram_list
    ram_list = []
    while True:
        for _ in range(500):  # Reducir aún más la cantidad de elementos añadidos
            ram_list.append(random.random())
        time.sleep(1)  # Aumentar la pausa para reducir el uso continuo de RAM

@app.route('/')
def index():
    return 'Hello, World!'

if __name__ == '__main__':
    # Crear y ejecutar hilos para consumir CPU y RAM
    threading.Thread(target=consume_cpu, daemon=True).start()
    threading.Thread(target=consume_ram, daemon=True).start()
    app.run(host='0.0.0.0', port=5000)

File: Proyecto1/containers/low_js/low.js
const express = require('express');
const app = express();
const PORT = process.env.PORT || 3000;

// Variables globales para almacenar el estado
let cpuResult = 0;
let ramLength = 0;

// Función para consumir CPU de manera menos intensiva
function consumeCPU() {
    setInterval(() => {
        let result = 0;
        for (let i = 0; i < 1000; i++) {  // Reducción en la cantidad de cálculos
            result += Math.sqrt(i) * Math.sin(i);
        }
        cpuResult = result;  // Actualiza la variable global
    }, 1000);  // Ejecutar cada 1000 ms (1 segundo)
}

// Función para consumir RAM de manera menos intensiva
function consumeRAM() {
    let ramArray = [];
    setInterval(() => {
        for (let i = 0; i < 50; i++) {  // Reducir aún más la cantidad de elementos
            ramArray.push(Math.random());
        }
        ramLength = ramArray.length;  // Actualiza la variable global
    }, 2500);  // Aumentar el intervalo de tiempo para reducir el uso de memoria
}

// Ruta principal
app.get('/', (req, res) => {
    res.send(`Hello, World! CPU: ${cpuResult}, RAM items: ${ramLength}`);
});

// Iniciar el servidor
app.listen(PORT, '0.0.0.0', () => {
    console.log(`Server is running on port ${PORT}`);
    // Ejecutar las funciones de consumo
    consumeCPU();
    consumeRAM();
});
